server/src/main/java/org/elasticsearch/action/admin/indices/settings/get/GetSettingsResponse.java

```diff
@@ -154,38 +144,44 @@ public String toString() {
         try {
             ByteArrayOutputStream baos = new ByteArrayOutputStream();
             XContentBuilder builder = new XContentBuilder(JsonXContent.jsonXContent, baos);
-            toXContent(builder, ToXContent.EMPTY_PARAMS, false);
+            var iterator = toXContentChunked(false);
```

Comment:
If we think this can grow to 10M+, should we then not avoid assembling this fully here, i.e., limit the output to say 100 entries?

Reply:
Fair point, not entirely trivial to do that, would have to add more logic to achieve that. I wonder if it matters all that much, we have the same problem for the other large responses as well.

I'll try to address this in the next PR on this while dropping the `toXContent` from these objects also. Incoming :)